sdk/identity/azure-identity/samples/client_certificate_credential.md

```diff
@@ -0,0 +1,69 @@
+
+```py
+credential = CertificateCredential(tenant_id, client_id, "./certs/cert.pfx")
+```
+Alternatively, the application can load the certificate themselves, such as in the following example.
+
+```py
+certificate_data = open(CERT_PATH, "rb").read()
+
+credential = new CertificateCredential(tenant_id, client_id, certificate_data=certificate_data)
+```
+
+## Rolling Certificates
+
+Long running applications may have the need to roll certificates during process execution. Certificate rotation is not currently supported by the `CertificateCredential` which treats the certificate used to construct the credential as immutable. This means that any clients constructed with a `CertificateCredential` using a particular cert would fail to authenticate requests after that cert has been rolled and the original is no longer valid. 
+
+However, if an application wants to roll this certificate without creating new service clients, it can accomplish this by creating its own `TokenCredential` implementation which wraps the `CertificateCredential`. The implementation of this custom credential `TokenCredential` would somewhat depend on how the application handles certificate rotation.
+
+### Explicit rotation
+
+If the application get's notified of certificate rotations and it can directly respond, it might choose to wrap the `CertificateCredential` in a custom credential which provides a means for rotating the certificate. 
+
+```py
+class RotatableCertificateCredential(object):
+    def __init__(self, tenant_id, client_id, certificate_path=None, **kwargs):
+        self._tenant_id = tenant_id
+        self._client_id = client_id
+        self._credential = CertificateCredential(tenant_id, client_id, certificate_path,**kwargs)
+    
+    def get_token(self, *scopes, **kwargs):
+        return self._credential.get_token(*scopes, **kwargs)
+
+    def rotate_certificate(certificate_path, **kwargs):
+        self._credential = CertificateCredential(self._tenant_id, self._client_id, certificate_path, **kwargs)
+```
+
+The above example shows a custom credential type `RotatableCertificateCredential` which provides a `rotate_certificate`. The implementation internally relies on a `CertificateCredential` instance `_credential`, and `rotate_certificate` simply replaces this instance with a new instance using the updated certificate.
+
+### Implicit rotation
+Some applications might want to respond to certificate rotations which are external to the application, for instance a separate process rotates the certificate by updating it on disk. Here the application create a custom credential which checks for certificate updates when tokens are requested. 
+
+```py
+class RotatingCertificateCredential(object):
+    def __init__(self, tenant_id, client_id, certificate_path=None, **kwargs):
+        self._tenant_id = tenant_id
+        self._client_id = client_id
+        self._certificate_path = certificate_path
+        self.refresh_certificate()
```

Comment:
I think we need to set an initial value for `self._certificate_last_modified` (using the current time should work, looking at the implementation)
```suggestion
        self._certificate_last_modified = time.time()
        self.refresh_certificate()
```